src/runtime_src/core/common/utils.cpp

```diff
@@ -259,4 +260,46 @@ parse_clock_id(const std::string& id)
   return clock_str != clock_map.end() ? clock_str->second : "N/A";
 }
 
+XRT_CORE_COMMON_EXPORT
+uint64_t
+mac_addr_to_value(const std::string& mac_addr){
+  std::string mac_addr_editable = mac_addr; // Make a copy to edit
```

Comment:
In such a case it is better to pass mac_addr by value and let the compiler do the copying.